apps/builder/app/builder/features/style-panel/shared/style-info.ts

```diff
@@ -428,11 +433,34 @@ export const useStyleInfo = () => {
     return getPresetStyleRule(component, tagName);
   }, [instances, selectedInstanceSelector, selectedInstanceIntanceToTag]);
 
+  const htmlStyle = useMemo(() => {
+    const instanceId = selectedInstanceSelector?.[0];
+    if (instanceId === undefined) {
+      return;
+    }
+    const tagName = selectedInstanceIntanceToTag?.get(instanceId);
```

Comment:
`tagName` still comes from the rendered DOM node :( 

Basically we can't rely on anything that comes from `SelectedInstanceConnector` if we want to allow to not render instances on canvas.

Not sure solving this is in scope of this PR, just an observation.

Reply:
It's not possible to solve unless data always provides currently used tag name